File: src/main/java/doom/despair/core/GameManagedObject.java
package doom.despair.core;

import java.util.UUID;

public abstract class GameManagedObject {
    private final UUID uniqueId = UUID.randomUUID();
    public final UUID getUUID() { return uniqueId; }
    @Override
    public final int hashCode() { return uniqueId.hashCode(); }
}

File: src/main/java/doom/despair/core/GameManagedObject.kt
package doom.despair.core

import java.util.*

abstract class GameManagedObject {
    val uUID: UUID = UUID.randomUUID()
    override fun hashCode(): Int {
        return uUID.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as GameManagedObject

        return uUID == other.uUID
    }
}
